src/snapshot.zig

```diff
@@ -0,0 +1,159 @@
+const std = @import("std");
+const base = @import("base.zig");
+const parse = @import("check/parse.zig");
+
+var verbose_log: bool = false;
+
+fn log(comptime fmt: []const u8, args: anytype) void {
+    if (verbose_log) {
+        std.log.info(fmt, args);
+    }
+}
+
+pub fn main() !void {
+    var gpa_impl = std.heap.GeneralPurposeAllocator(.{}){};
+    defer {
+        _ = gpa_impl.deinit();
+    }
+    const gpa = gpa_impl.allocator();
```

Comment:
super nit: just noting a best practice, but doesn't really matter for this side script

also make an arena use it for args along with `allocPrint` and things of that nature.